src/main/java/utils/TaskList.java

```diff
@@ -0,0 +1,180 @@
+package utils;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import duke.Duke;
+import entities.Deadline;
+import entities.Event;
+import entities.Task;
+import entities.Todo;
+import exceptions.DukeException;
+
+/**
+ * Stores and manages the tasks entered by the user.
+ */
+public class TaskList {
+    /**
+     * The list of tasks entered by the user.
+     */
+    private List<Task> tasks;
+
+    /**
+     * Constructor for a TaskList instance.
+     */
+    public TaskList() {
+        this.tasks = new ArrayList<>();
+    }
+
+    /**
+     * Constructor for a TaskList instance.
+     *
+     * @param taskList The List of tasks loaded from cache.
+     * @throws DukeException If no cached tasks are available.
+     */
+    public TaskList(List<Task> taskList) throws DukeException {
+        if (taskList.isEmpty()) {
+            throw new DukeException("No cached tasks.");
+        } else {
+            this.tasks = taskList;
+            Ui.sendMessage("Tasks loaded from cache.");
+        }
+    }
+
+    /**
+     * Getter for length of task List.
+     * @return
+     */
+    public int getSize() {
+        return this.tasks.size();
+    }
+
+    /**
+     * Utility function with logic for adding tasks to the user's task list.
+     * @param description The description of the task to be added.
+     * @param type The type of task to be added.
+     * @param remarks The remarks to be added for events or deadlines.
+     */
+    public void add(String description, Duke.TaskType type, String remarks) {
+        String s = "Got it. I've added this task:\n\t";

```

Comment:
Could the variable `s` be renamed something clearer like `message`? It's scope is quite large for it to be a scratch variable.